lib/coreboot/cbmem_console.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2017, ARM Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <coreboot/cbmem_console.h>
+#include <xlat_tables_v2.h>
+
+/*
+ * The console is initially validated by coreboot, so we can trust it. But at
+ * runtime the non-secure OS could manipulate the size field to trick us into
+ * making bad memory accesses. Therefore we need to store the known-good size
```

Comment:
If this memory is accessible to the non-secure OS, then having this mapped into BL31 at runtime is a security attack vector. I'll have to trust that you've thought through all the implications of this.